project1/main.cpp

```diff
@@ -41,43 +51,107 @@ int main() {
         DayCounter dayCounter = Actual365Fixed();
         Handle<YieldTermStructure> riskFreeRate(
             ext::shared_ptr<YieldTermStructure>(
-                new ZeroCurve({today, today + 6*Months}, {0.01, 0.015}, dayCounter)));
+                new ZeroCurve({ today, today + 6 * Months }, { 0.01, 0.015 }, dayCounter)));
         Handle<BlackVolTermStructure> volatility(
             ext::shared_ptr<BlackVolTermStructure>(
-                new BlackVarianceCurve(today, {today+3*Months, today+6*Months}, {0.20, 0.25}, dayCounter)));
+                new BlackVarianceCurve(today, { today + 3 * Months, today + 6 * Months }, { 0.20, 0.25 }, dayCounter)));
```

Comment:
About the fact that the NPV doesn't change: here the first point is at 3 months, which is the maturity of the option, so the volatility is already constant before that.  If you use 
```suggestion
                new BlackVarianceCurve(today, { today + 2 * Months, today + 6 * Months }, { 0.20, 0.25 }, dayCounter)));
```
instead, you should see the NPV change.